src/auto_contiguous.cpp

```diff
@@ -33,43 +33,60 @@ inline namespace MIGRAPHX_INLINE_NS {
 
 void auto_contiguous::apply(module& m) const
 {
-    std::string key = "require_std_shape";
-    for(auto ins : reverse_iterator_for(m))
-    {
-        auto&& attr = ins->get_operator().attributes();
-        if((attr.get(key, false)))
-        {
-            auto args     = ins->inputs();
-            auto new_args = args;
-            std::transform(args.begin(), args.end(), new_args.begin(), [&](auto in) {
-                if(in->name() == "contiguous")
-                {
-                    return in;
-                }
-                return m.insert_instruction(ins, make_op("contiguous"), in);
-            });
+    // std::string key = "require_std_shape";
+    // for(auto ins : reverse_iterator_for(m))
+    // {
+    //     auto&& attr = ins->get_operator().attributes();
+    //     if((attr.get(key, false)))
+    //     {
+    //         auto args     = ins->inputs();
+    //         auto new_args = args;
+    //         std::transform(args.begin(), args.end(), new_args.begin(), [&](auto in) {
+    //             if(in->name() == "contiguous")
+    //             {
+    //                 return in;
+    //             }
+    //             return m.insert_instruction(ins, make_op("contiguous"), in);
+    //         });
 
-            if(new_args != args)
-            {
-                m.replace_instruction(ins, ins->get_operator(), new_args);
-            }
-        }
-    }
+    //         if(new_args != args)
+    //         {
+    //             m.replace_instruction(ins, ins->get_operator(), new_args);
+    //         }
+    //     }
+    // }
 
     auto last = std::prev(m.end());
     for(auto ins : iterator_for(m))
     {
-        if(ins->name() == "layout")
+        if(ins->name() == "layout" or ins->name() == "contiguous")
             continue;
         // for last instruction that is NOT a return
         if(ins->outputs().empty() and ins != last)
             continue;
-        shape s = ins->get_shape();
-        if(not s.dynamic() and not s.standard() and s.elements() != 0)
+        auto args     = ins->inputs();
+        auto new_args = args;
+        std::transform(args.begin(), args.end(), new_args.begin(), [&](auto in) {
+            if(in->name() == "contiguous" or in->name() == "layout")
+                return in;
+            if(in->name() == "@literal" and in->get_shape().standard())
+                return in;
+            return m.insert_instruction(ins, make_op("contiguous"), in);
+        });
+
+        if(new_args != args)
         {
-            auto c = m.insert_instruction(std::next(ins), make_op("contiguous"), ins);
-            m.replace_instruction(ins, c);
+            if(not ins->module_inputs().empty())
+                m.replace_instruction(ins, ins->get_operator(), new_args, ins->module_inputs());
+            else
+                m.replace_instruction(ins, ins->get_operator(), new_args);
```

Comment:
You shouldn't replace the input arguments as that can cause multiple contiguous operators to be inserted when its used in multiple places. This should just do:

```cpp
if(contains({"layout", "contigous"}, ins->name()))
    continue;
// for last instruction that is NOT a return
if(ins->outputs().empty() and ins != last)
    continue;
shape s = ins->get_shape();
if(s.dynamic())
    continue;
if(s.type() == shape::tuple_type)
    continue;
if(s.standard() and ins->name() == "@literal")
    continue;
auto c = m.insert_instruction(std::next(ins), make_op("contiguous"), ins);
m.replace_instruction(ins, c);
```